new_course/1_programming_fundementals/problem_solving.md

```diff
@@ -1,31 +1,274 @@
+
+It's important to remember that you might not know all the steps that you might need up front so your algorithm may be incomplete, this is fine. Getting started with and solving one of the sub problems you have identified in the planning stage, often reveals the next sub problem you can work on. Or if you already know the next sub problem is it’s often simpler with the first sub problem solved.
+
+Many beginners try to solve the big problem in one go and. **Don’t do this**. If the problem is sufficiently complex enough you’ll get yourself tied in knots and make life a lot harder for yourself. Decomposing problems into smaller easier to solve sub problems is a much better approach as decomposition is the main way of dealing with complexity and making problems easier and more approachable to solve and understand.
+
+In short, break the big problem down and solve each of the smaller problems until you've solved the big problem.
+
+### Solving Fizz Buzz
+To demonstrate this workflow in action lets solve a common programming exercise, [FizzBuzz](https://en.wikipedia.org/wiki/Fizz_buzz)
+
+#### Understanding The Problem
+> Write a program takes a users input and prints the numbers from one to the number the user inputed. But for multiples of three print "Fizz" instead of the number and for the multiples of five print "Buzz". For numbers which are multiples of both three and five print "FizzBuzz".
+
+This is the big picture problem we will be solving. It's pretty simple so we may not need to reword it. But we can always make it clearer by rewording it.
+
+Write a problem that allows the user to enter a number, Print each number between one the number the user entered, but for numbers that divide by three cleanly without a remainder print "Fizz" instead. For numbers that divide by 5 without a remainder print "Buzz" and finally for numbers that divide by both three and five without a remainder print "FizzBuzz".
+
+#### Plan
+Does your program have an interface? what will it look like?
+Our FizzBuzz solution will be a command line program, so we don't need an interface. The only user interaction will be allowing users to enter a number.
+
+What inputs will your program have? will the user enter data or will you get input from somewhere else
+The user will enter a number from the command line.
+
+Whats the desired output?
+The desired output is a list of numbers from 1 to the number the user entered. But each number that is divisible by 3 will output Fizz, each number that is divisible by 5 will output "Buzz" and each number that is divisible by both 3 and 5 will output "FizzBuzz"
+
+Given your inputs, what are the steps necessary to return the desired output?
+The algorithm in pseudo code for this problem:
+
+```
+When a user inputs a number
+Loop from 1 to the entered number
+Print each number
```

Comment:
Print each number
But if the number is divisible by 3 or 5 the number doesn't get printed? This should go  at `line 89` with something like `Otherwise print the number`.